### PR TITLE
STORM-3555 add meter to track errors killing workers

### PR DESCRIPTION
**Pull request by agresch**

### CODE REVIEW

File: storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java [hunk 1]
```diff
@@ -339,6 +342,7 @@ public void launchDaemon() {
             //This will only get updated once
             metricsRegistry.registerMeter("supervisor:num-launched").mark();
             metricsRegistry.registerMeter("supervisor:num-shell-exceptions", ShellUtils.numShellExceptions);
+            killErrorMeter = metricsRegistry.registerMeter("supervisor:num-kill-errors");
```
Comment (govind-menon) [inline storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java]: Nit: rename to num-kill-worker-errors

Comment (agresch) [reply storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java]: @govind-menon - renamed

File: storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java [hunk 2]
```diff
@@ -528,6 +532,9 @@ void killWorkers(Collection<String> workerIds, ContainerLauncher launcher) throw
                 long start = Time.currentTimeMillis();
                 while (!k.areAllProcessesDead()) {
                     if ((Time.currentTimeMillis() - start) > 10_000) {
+                        if (killErrorMeter != null) {
+                            killErrorMeter.mark();
+                        }
```
Comment (Ethanlm) [inline storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java]: If this happens, the supervisor will exit. What's the value of this meter

Comment (agresch) [reply storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java]: Line 545 in the modified version of this file catches the exception.  We just log an error.

Comment (Ethanlm) [reply storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java]: What's the modified version?

Comment (agresch) [reply storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java]: Supervisor.java that I modified, or line 539 of Supervisor.java on master currently.

Comment (Ethanlm) [reply storm-server/src/main/java/org/apache/storm/daemon/supervisor/Supervisor.java]: Ah missed that. Thanks

**Comment by Ethanlm** [comment]
Looks good to me. Could you please rebase so it can pick up the latest travis.yaml file and so it can run a travis build?

**Comment by agresch** [comment]
@Ethanlm  - rebased

**Pull request merged**